include/complex.h

```diff
@@ -0,0 +1,271 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libphoenix
+ *
+ * mathematical functions on complex numbers
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Gerard Swiderski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#ifndef _COMPLEX_H_
+#define _COMPLEX_H_
+
+/* define complex numbers specifers */
+#define complex    _Complex
+#define _Complex_I 1.0fi
+#define I          _Complex_I
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+static inline double creal(double complex z)
+{
+	return (__real__(z));
+}
+
+
+static inline float crealf(float complex z)
+{
+	return (__real__(z));
+}
+
+
+static inline double cimag(double complex z)
+{
+	return (__imag__(z));
+}
+
+
+static inline float cimagf(float complex z)
+{
+	return (__imag__(z));
+}
+
+
+static inline double complex conj(double complex z)
+{
+	union {
+		double complex z;
+		struct {
+			double re;
+			double im;
+		} part;
+	} u = { .z = z };
+	u.part.im = -u.part.im;
+	return u.z;
+}
+
+
+static inline float complex conjf(float complex z)
+{
+	union {
+		float complex z;
+		struct {
+			float re;
+			float im;
+		} part;
+	} u = { .z = z };
```

Comment:
Uhh, I don't really like this union, specifically assuming _Complex implementation, on which we have no control.